microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -2,47 +2,316 @@
+    requirements=[
+        OsPackageDependencies(
+            os_matcher=lambda x: isinstance(x, Ubuntu)
+            and x.information.codename == "bionic",
+            packages=[
+                "build-essential",
+                "libmnl-dev",
+                "libelf-dev",
+                "libnuma-dev",
+                "dpkg-dev",
+                "pkg-config",
+                "python3-pip",
+                "python3-pyelftools",
+                "python-pyelftools",
+                # 18.04 doesn't need linux-modules-extra-azure
+                # since it will never have MANA support
+            ],
+        ),
+        OsPackageDependencies(
+            os_matcher=lambda x: isinstance(x, Debian),
+            packages=[
+                "build-essential",
+                "libnuma-dev",
+                "libmnl-dev",
+                "python3-pyelftools",
+                "libelf-dev",
+                "pkg-config",
+                "linux-modules-extra-azure",
+            ],
+        ),
+        OsPackageDependencies(
+            os_matcher=lambda x: isinstance(x, Suse),
+            packages=[
+                "psmisc",
+                "libnuma-devel",
+                "numactl",
+                "libmnl-devel meson",
+                "gcc-c++",
+            ],
+        ),
+        OsPackageDependencies(
+            os_matcher=lambda x: isinstance(x, (Fedora)),
+            packages=[
+                "psmisc",
+                "numactl-devel",
+                "pkgconfig",
+                "elfutils-libelf-devel",
+                "python3-pip",
+                "kernel-modules-extra",
+                "kernel-headers",
+                "gcc-c++",
+            ],
+        ),
+    ]
+)
+
+
+def get_debian_backport_repo_args(os: Debian) -> List[str]:
+    if not isinstance(os, Debian):
+        return []
+    if isinstance(os, Ubuntu) and (
+        is_ubuntu_latest_or_prerelease(os) or not is_ubuntu_lts_version(os)
+    ):
+        return []
+    repos = os.get_repositories()
+    backport_repo = f"{os.information.codename}-backports"
+    if any([backport_repo in repo.name for repo in repos]):
+        return [f"-t {backport_repo}"]
+    return []
+
+
+class DpdkPackageManagerInstall(PackageManagerInstall):
+    def _setup_node(self) -> None:
+        if isinstance(self._os, Debian):
+            self._package_manager_extra_args = get_debian_backport_repo_args(self._os)
+            if self._os.information.version < "22.4.0":
+                self._os.update_packages("linux-azure")
+                self._node.reboot()
+
+    def get_installed_version(self) -> VersionInfo:
+        return self._os.get_package_information("dpdk", use_cached=False)
+
+    def _check_if_installed(self) -> bool:
+        return self._os.package_exists("dpdk")
+
+
+# implement SourceInstall for DPDK
+class DpdkSourceInstall(Installer):
+    _dpdk_devname_repo = "https://www.github.com/mcgov/devname.git"
```

Comment:
we need to avoid personal repos.

Reply:
I'm not maintaining two copies of the project, I'll update the link when it's checked into the DPDK project.

Reply:
What means "link"? It looks a core c file, please either include the c file in LISA repo, or create a Microsoft repo (not under personal name). I prefer to maintain the file in LISA. You can still use it in other places. Because the extra repo will be hold by our team too.

Reply:
`_dpdk_devname_repo = "https://www.github.com/mcgov/devname.git" ` <- the link

Sure. For now, I'll just remove it since it's not in use yet and I'm going to change how it's implemented when we start to use it.

Reply:
Resolving since this was removed.